### PR TITLE
fix admin sales

### LINKED ISSUE #1807: Admin sales tab not working
Admin sales tab just has dummy data.

screenshot from 2019-01-14 21-53-10

### PR DESCRIPTION
**Pull request by YogeshSharma0201**
This PR requires backend changes (fossasia/open-event-server/pull/5528) to be merged first.
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
- Admin sales is linked to backend now and doesn't use dummy data.

Fixes #1807

### CODE REVIEW

File: app/routes/admin/sales/organizers.js [hunk 1]
```diff
@@ -5,35 +5,12 @@ export default Route.extend({
...
-        placedSales      : 1,
-        pendingTickets   : 1,
-        pendingSales     : 1
-      }
-    ];
+  async model() {
+    let filterOptions = [];
+
+    return this.store.query('admin-sales-by-organizer', {
+      filter       : filterOptions,
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/organizers.js]: you may remove the filterOptions, as there is no filter required.

File: app/routes/admin/sales/organizers.js [hunk 2]
```diff
@@ -5,35 +5,12 @@ export default Route.extend({
...
+      'page[size]' : 10
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/organizers.js]: Don't limit the page size to 10, rather for now define it as `0` to fetch all the records as admin sales table does not has pagination yet. If you like, you can work on converting it to an ember model table after this PR is merged.

File: app/routes/admin/sales/marketer.js [hunk 3]
```diff
@@ -4,17 +4,13 @@ export default Route.extend({
...
-      tickets   : 500,
-      sales     : 1000.00,
-      discounts : 200.00
-    }];
+
+  async model() {
+    let filterOptions = [];
+
+    return this.store.query('admin-sales-by-marketer', {
+      filter       : filterOptions,
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/marketer.js]: you may remove the filterOptions, as there is no filter required.

File: app/routes/admin/sales/locations.js [hunk 4]
```diff
@@ -4,23 +4,13 @@ export default Route.extend({
...
-      placedSales      : 32.00,
-      pendingTickets   : 4,
-      pendingSales     : 32.00
-    }];
+
+  async model() {
+    let filterOptions = [];
+
+    return this.store.query('admin-sales-by-location', {
+      filter       : filterOptions,
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/locations.js]: you may remove the filterOptions, as there is no filter required.

File: app/routes/admin/sales/index.js [hunk 5]
```diff
@@ -1,28 +1,16 @@
...
-      placedTickets    : 4,
-      placedSales      : 32.00,
-      pendingTickets   : 4,
-      pendingSales     : 32.00
-    }];
+  async model() {
+    let filterOptions = [];
+
+    return this.store.query('admin-sales-by-event', {
+      filter       : filterOptions,
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/index.js]: you may remove the filterOptions, as there is no filter required.

File: app/routes/admin/sales/fees.js [hunk 6]
```diff
@@ -5,17 +5,12 @@ export default Route.extend({
...
-      event   : 'Sample Event2',
-      tickets : 3,
-      fee     : 10,
-      revenue : 40
-    }];
+  async model() {
+    let filterOptions = [];
+
+    return this.store.query('admin-sales-fee', {
+      filter       : filterOptions,
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/fees.js]: you may remove the filterOptions, as there is no filter required.

File: app/routes/admin/sales/discounted-events.js [hunk 7]
```diff
@@ -5,43 +5,12 @@ export default Route.extend({
...
-        completedTickets : 6,
-        completedSales   : 2.30,
-        placedTickets    : 40,
-        placedSales      : 32.00,
-        pendingTickets   : 42,
-        pendingSales     : 36.00
-      }
-    ];
+    return this.store.query('admin-sales-discounted', {
+      filter       : filterOptions,
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/discounted-events.js]: you may remove the filterOptions, as there is no filter required.

File: app/routes/admin/sales/marketer.js [hunk 8]
```diff
@@ -4,17 +4,13 @@ export default Route.extend({
...
+      'page[size]' : 10
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/marketer.js]: Don't limit the page size to 10, rather for now define it as 0 to fetch all the records as admin sales table does not has pagination yet. If you like, you can work on converting it to an ember model table after this PR is merged.

File: app/routes/admin/sales/locations.js [hunk 9]
```diff
@@ -4,23 +4,13 @@ export default Route.extend({
...
+      'page[size]' : 10
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/locations.js]: Don't limit the page size to 10, rather for now define it as 0 to fetch all the records as admin sales table does not has pagination yet. If you like, you can work on converting it to an ember model table after this PR is merged.

File: app/routes/admin/sales/index.js [hunk 10]
```diff
@@ -1,28 +1,16 @@
...
+      'page[size]' : 10
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/index.js]: Don't limit the page size to 10, rather for now define it as 0 to fetch all the records as admin sales table does not has pagination yet. If you like, you can work on converting it to an ember model table after this PR is merged.

File: app/routes/admin/sales/fees.js [hunk 11]
```diff
@@ -5,17 +5,12 @@ export default Route.extend({
...
+      'page[size]' : 10
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/fees.js]: Don't limit the page size to 10, rather for now define it as 0 to fetch all the records as admin sales table does not has pagination yet. If you like, you can work on converting it to an ember model table after this PR is merged.

File: app/routes/admin/sales/discounted-events.js [hunk 12]
```diff
@@ -5,43 +5,12 @@ export default Route.extend({
...
+      'page[size]' : 10
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/discounted-events.js]: Don't limit the page size to 10, rather for now define it as 0 to fetch all the records as admin sales table does not has pagination yet. If you like, you can work on converting it to an ember model table after this PR is merged.

File: app/routes/admin/sales/discounted-events.js [hunk 13]
```diff
@@ -5,43 +5,11 @@ export default Route.extend({
...
-        marketer         : 'abc@gmail.com',
-        completedTickets : 1,
-        completedSales   : 0.00,
-        placedTickets    : 1,
-        placedSales      : 0.00,
-        pendingTickets   : 1,
-        pendingSales     : 0.00
-      },
+  async model() {
+    let filterOptions = [];
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/discounted-events.js]: you can get rid of filterOptions completely.

File: app/routes/admin/sales/discounted-events.js [hunk 14]
```diff
@@ -5,43 +5,11 @@ export default Route.extend({
...
-        marketer         : 'sample@gmail.com',
-        completedTickets : 6,
-        completedSales   : 2.30,
-        placedTickets    : 40,
-        placedSales      : 32.00,
-        pendingTickets   : 42,
-        pendingSales     : 36.00
-      }
-    ];
+    return this.store.query('admin-sales-discounted', {
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/discounted-events.js]: use `this.store.findAll()`

**Comment by shreyanshdwivedi** [comment]
@YogeshSharma0201 great work :tada:

File: app/routes/admin/sales/discounted-events.js (hunk 14 above)
Comment (YogeshSharma0201) [reply app/routes/admin/sales/discounted-events.js]: @CosmicCoder96 i have made the changes.

**Pull request merged**